sdcm/utils/azure_region.py

```diff
@@ -0,0 +1,425 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2021 ScyllaDB
+
+from __future__ import annotations
+
+import os
+import enum
+import logging
+import binascii
+from typing import TYPE_CHECKING
+from functools import cached_property
+from contextlib import suppress
+
+from azure.core.exceptions import ResourceNotFoundError
+from azure.mgmt.compute.models import TargetRegion
+
+from sdcm.utils.azure_utils import AzureService
+
+if TYPE_CHECKING:
+    # pylint: disable=ungrouped-imports
+    from typing import Optional
+
+    from azure.mgmt.compute.models import Gallery, GalleryImageVersion, VirtualMachine
+    from azure.mgmt.network.models import (
+        NetworkInterface,
+        NetworkSecurityGroup,
+        PublicIPAddress,
+        Subnet,
+        VirtualNetwork,
+    )
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AzureOsState(enum.Enum):
+    GENERALIZED = "Generalized"
+    SPECIALIZED = "Specialized"
```

Comment:
Would be really useful to have some comments in lots of such places that is specific for Azure.
What is the difference between these states? Why should we care about it?

Reply:
I've added comments on that